app/ui/design-system/src/lib/Components/AppLink.tsx

```diff
@@ -0,0 +1,42 @@
+import { Link } from "@remix-run/react"
+import React from "react"
+
+export function isLinkExternal(url: string) {
+  return /^(https?:\/\/|www\.)/.test(url)
+}
+
+const AppLink = ({
+  to,
+  className,
+  title,
+  style,
+  children,
+}: {
+  to: string
+  title?: string
+  className?: string
+  style?: React.CSSProperties
+  children: React.ReactNode
+}) => {
+  const classes = className || "hover:opacity-75"
+  if (isLinkExternal(to))
+    return (
+      <a
+        href={to}
+        target="blank"
+        rel="noreferrer"
+        className={classes}
+        title={title}
+      >
+        {children}
+      </a>
+    )
+
+  return (
+    <Link to={to} className={classes} title={title} style={style}>
+      {children}
+    </Link>
+  )
+}
```

Comment:
random TS trivia, but an alternative type for the props on this that's might be spreadable is something like


```tsx
const AppLink = ({to, ...props}: Omit<JSX.IntrinsicElements['a'], "href"> & {to: string}) => {
  if (isLinkExternal(to)) {
    return <a href={to} {...props} />
  } else {
    return <Link to={to} {...props} />
  }
}
```

Reply:
`Omit` !

Reply:
Nice! I went with explicit props because `a` and `Link` have different types. Could also reverse it to a Pick<...similar attributes>